posts/2017-09-05-three-types-of-strings.md

```diff
@@ -0,0 +1,106 @@
+---
+layout: post
+title: There are at least three types of strings
+
+meta:
+  nav: blog
+  author: S11001001
+  pygments: true
+---
+
+[Newtype mechanisms](https://contributors.scala-lang.org/t/pre-sip-unboxed-wrapper-types/987)
+are a great way to introduce wrapper-free, global distinctions of
+“different” instances of the same type. But we can do that on a local
+level, too,
+[by using type parameters](https://gist.github.com/jbgi/d6b677d084fafc641fe01f7ffd00591c#file-label-java-L32).
```

Comment:
You can use a [permanent link](https://gist.github.com/jbgi/d6b677d084fafc641fe01f7ffd00591c/70842ca600e53e8c237c681773fe4e16bd679628#file-label-java-L32) here.

Reply:
Thanks for the tip; I couldn't find it! Added in 08213cd.